inbox/util/threading.py

```diff
@@ -16,25 +19,24 @@ def fetch_corresponding_thread(db_session, namespace_id, message):
     # to a message always has a similar subject. This is only
     # right 95% of the time.
     clean_subject = cleanup_subject(message.subject)
+
+    # XXX: It is much faster to sort client-side by message ID, we therefore
```

Comment:
Another option would be the break this apart into two queries and first get a list of thread ids and then for each thread query the messages. This would let us apply limits to both and let ordering be done in the DB. Using a `limit` and then sorting client side made me think about this since that approach could end up with different results since the `limit` is applied before the sorting.

Reply:
> that approach could end up with different results since the `limit` is applied before the sorting

Only within threads. We could skip the last thread if you're concerned.

Reply:
> XXX: It is much faster to sort client-side by message ID

What do you mean by "by message ID"? We order by message's `received_date` below.

Reply:
corrected in d0b1d1239a2096bdbe95efe3c96ad893149a34fc